### PR TITLE
fix: add safari polyfill for nested web workers

### PR DESCRIPTION
**Pull request by olehrakwix**

### CODE REVIEW

File: packages/runtime-node/src/worker-thread-entry.ts [hunk 1]
```diff
@@ -55,13 +54,13 @@ const handleStartupMessage = async (command: WorkerThreadStartupCommand) => {
...
     disposables.add(() => {
-        parentPort!.off('message', messageHandler);
+        worker.removeEventListener('message', messageHandler);
         return engine.shutdown();
     });
 };
 
-const messageHandler = (message: unknown) => {
-    const workerThreadCommand = message as WorkerThreadCommand;
+const messageHandler = (message: any) => {
```
Comment (barak007) [inline packages/runtime-node/src/worker-thread-entry.ts:62]: why this changed to any?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -15,8 +15,11 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
+          registry-url: 'https://registry.npmjs.org/'
```
Comment (barak007) [inline .github/workflows/tests.yml:18]: why do we need this?

**Review by barak007 — COMMENTED** [review]
Looks good. few I left couple of questions.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (olehrakwix) [reply .github/workflows/tests.yml:18]: Cause @wixc3/isomorphic-worker lives there, and I got these errors without this change:
https://github.com/wixplosives/engine/actions/runs/5110105664/jobs/9185624849#step:6:13

File: packages/runtime-node/src/worker-thread-entry.ts (hunk 1 above)
Comment (olehrakwix) [reply packages/runtime-node/src/worker-thread-entry.ts]: changed back to `unknown`

File: packages/core/src/com/hosts/universal-worker-host.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import type {
+    UniversalMessageHandler,
+    UniversalWorker,
+    UniversalWorkerUserMethods,
+} from '@wixc3/isomorphic-worker/dist/types';
```
Comment (AviVahl) [inline packages/core/src/com/hosts/universal-worker-host.ts:5]: avoid importing directly from dist

**Review by AviVahl — CHANGES_REQUESTED** [review]
the approach here uses a private package

File: packages/core/src/com/hosts/universal-worker-host.ts (hunk 3 above)
Comment (olehrakwix) [reply packages/core/src/com/hosts/universal-worker-host.ts]: fixed

**Review by AviVahl — APPROVED** [review]

**Pull request merged**